wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```
@@ -0,0 +1,251 @@
+	 * @throws      \InvalidArgumentException       if the value contains objects, which are not an instance of UploadFile
+	 */
+	public function value($value) {
+		if (!is_array($value)) {
+			throw new \InvalidArgumentException('$value must be an array.');
+		}
+		
+		foreach ($value as $file) {
+			if (!($file instanceof UploadFile)) {
+				throw new \InvalidArgumentException('All given files must be an instance of '. UploadFile::class .'.');
+			}
+		}
+		
+		$this->registerField();
+		
+		UploadHandler::getInstance()->registerFilesByField($this->getId(), $value);
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function hasSaveValue() {
+		return false;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function cleanup() {
+		$this->unregisterField();
+		
+		return $this;
+	}
+	
+	/**
+	 * @inheritDoc
+	 * 
+	 * @throws      \RuntimeException       if the field has already been initalisated
+	 */
+	public function maximum($maximum = null) {
+		if ($this->isRegistered()) {
+			throw new \LogicException('The upload field has already been registered. Therefore no modifications are allowed.');
+		}
+		
+		return $this->traitMaximum($maximum);
+	}
+	
+	/**
+	 * Sets the imageOnly flag for this field.
+	 *
+	 * @param	boolean	        $imageOnly
+	 * @return	static				this field
+	 */
+	public function imageOnly($imageOnly = true) {
+		$this->imageOnly = $imageOnly;
+		
+		return $this;
+	}
+	
+	/**
+	 * Sets the imageOnly flag for this field.
+	 *
+	 * @param	boolean	        $allowSvgImages
+	 * @return	static				this field
+	 */
+	public function allowSvgImage($allowSvgImages = true) {
+		$this->allowSvgImage = $allowSvgImages;
+		
+		return $this;
+	}
+	
+	/**
+	 * Returns true, if the field is an image only field (only images can be uploaded).
+	 *
+	 * @return	boolean
+	 */
+	public function isImageOnly() {
+		return $this->imageOnly;
+	}
+	
+	/**
+	 * Returns true, if the field can contain svg images.
```

This method is called `svgImagesAllowed` (plural)  in `UploadField`. Please also document that this setting is only relevant if only images are allowed. May-be `allowSvgImage()` should throw an exception if `$this->imageOnly = false`?